definitions/npm/redux-form_v5.x.x/flow_v0.53.x-/redux-form_v5.x.x.js

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+
+declare module 'redux-form' {
+  declare export type InputProps = {
+    name: string,
+    value: string | boolean,
+    valid: boolean,
+    invalid: boolean,
+    dirty: boolean,
+    pristine: boolean,
+    active: boolean,
+    touched: boolean,
+    visited: boolean,
+    autofilled: boolean,
+    error?: string,
+    onChange: (eventOrValue: SyntheticEvent<any> | string | boolean) => mixed,
+    onUpdate: (eventOrValue: SyntheticEvent<any>  | string | boolean) => mixed,
+    onBlur: (eventOrValue: SyntheticEvent<any>  | string | boolean) => mixed,
+    onDragStart: Function,
+    onDrop: Function,
+    onFocus: Function,
+  };
+
+  declare export type MetaProps = {
+    active: boolean,
+    autofilled: boolean,
+    asyncValidating: boolean,
+    dirty: boolean,
+    dispatch: Function,
+    error?: string,
+    invalid: boolean,
+    pristine: boolean,
+    submitting: boolean,
+    touched: boolean,
+    valid: boolean,
+    visited: boolean,
+  };
+
+  declare export type FieldInputProps = {
+    input: InputProps,
+    meta: MetaProps,
+  };
+
+  declare type FunctionComponent<P, S> = (props?: P, void) => ?React$Element<any>;
+  declare type ClassComponent<P, S> = Class<React$Component<P, S>>;
+  declare type Component<P> = FunctionComponent<P, any> | ClassComponent<P, any>;
```

Comment:
I think all of these three could be replaced with builtin react types: https://flow.org/en/docs/react/types/